### ISSUE TITLE
feat: less-verbose parameters syntax with positional-args support

### ISSUE THREAD

**Issue by mumoshu** [mumoshu/variant#88]
TL;DR; Use YAML hash for `parameters`(variant already has it implemented, just recommend its usage) and add `usage` setting for setting up positional args.

---

`variant` has been using an YAML array `parameters`, each array item has `name`. Due to the nature of parameters it doesn't allow duplicates.

The valid question would be - why variant` used the array syntax for parameters? That's for positional argument.

Historically, variant already has a "hidden"(?) feature that treats `parameters` as a YAML hash, each item optionally having`argument-index: N`, where N is an integer originating from 0, that treats the parameter as a positional argument at N:

```yaml
tasks:
  foo:
    parameters:
      bar:
        type: string
        description: "the bar"
        argument-index: 0
      environment:
        type: string
        default: "heaven"
    script: |
      echo "Hello {{ get "bar" }} you are in the {{ get "environment" }}"
```

I didn't really like `argument-index`. That's why variant uses the following syntax with an array as parameters, where each array item at index N is automatically treated as Nth argument:

```yaml
tasks:
  foo:
    parameters:
    - name: bar
      type: string
      description: "the bar"
    - name: environment
      type: string
      default: "heaven"
    script: |
      echo "Hello {{ get "bar" }} you are in the {{ get "environment" }}"
```

So I propose going back to hashes, while removing `argument-index` and adding `usage` instead for configuring positional args:

```yaml
tasks:
  foo:
    usage: `foo [flags] BAR`
    parameters:
      bar:
        type: string
        description: "the bar"
      environment:
        type: string
        default: "heaven"
    script: |
      echo "Hello {{ get "bar" }} you are in the {{ get "environment" }}"
```

The `usage` has the syntax very similar to that produced by `cobra`. For example `helm` produces usages like this:

```
$ helm status --help

Usage:
  helm status [flags] RELEASE_NAME
```

The idea of `usage` is to instruct the user to explicitly specify the "Usage" output that is used instead of the one produced by `cobra`. Variant parses the user-specified usage, and configures `cobra` accordingly, so that `cobra` accepts positional-args as the user wanted and produces Usage outputs as the user wanted.

**Comment by mumoshu** [comment]
Closing this in favor of #109. Thanks @Nuru for the great suggestion!
